apps/trackers/bts_tracker.py

```diff
@@ -0,0 +1,157 @@
+
+    def _generate_summary(self) -> str:
+        """
+        Generates the summary of a tracker
+        """
+        summary = f"{self._flaw.cve_id} {self._affect.ps_component}: {self._flaw.title}"
+        if self._stream:
+            summary += f" [{self._stream.name}]"
+        return summary
+
+    def _generate_labels(self) -> list[str]:
+        """
+        Generates labels used by Jira to filter trackers
+        """
+        labels = []
+        if self._affect.ps_component:
+            labels.append(f"pscomponent:{self._affect.ps_component}")
+        labels.append(self._flaw.cve_id)
+        labels.append("Security")
+        labels.append("SecurityTracking")
+
+        return labels
+
+    def _generate_sla(self):
+        # TODO OSIDB-96
+        return ""
+
+    def _impact_to_priority(self) -> JiraPriority:
+        """
+        Convert OSIDB impact ot Jira Priority
+        """
+        allowed_values = JiraProjectFields.objects.get(
+            project_key=self._ps_module.bts_key, field_id="priority"
+        ).allowed_values
+        allowed_values = [value["name"] for value in allowed_values]
+        for priority in IMPACT_TO_JIRA_PRIORITY[self._flaw.impact]:
+            if priority in allowed_values:
+                return priority
+
+        raise NoPriorityAvailableError(
+            f"Jira project {self._ps_module.bts_key} does not have a corresponding priority for impact "
+            f"{self._flaw.impact}; allowed Jira priority values are: {', '.join(allowed_values)}"
+        )
+
+    def _generate_description(self) -> str:
+        """
+        Generates a text description for the bug being tracked
+        """
+        if self._flaw.is_embargoed:
+            description = "Security Tracking Issue\n\nDo not make this issue public.\n"
+        else:
+            description = "Public Security Tracking Issue\n"
+
+        description += f"OSIDB Flaw: {self._flaw.uuid}.\n"
+
+        if self._stream and self._stream.rhsa_sla_applicable:
+            description += f"{self._generate_sla()}\n"
+        return description
+
+    def generate_bts_object(self) -> Dict[str, Any]:
+        """
+        Generates an object that can be used to create a new Jira tracker
+        """
+        bts_tracker = {
+            "fields": {
+                "project": {"key": self._ps_module.bts_key},
+                "issuetype": {"name": "Bug"},
+                "priority": {"name": self._impact_to_priority()},
+                "summary": self._generate_summary(),
+                "description": self._generate_description(),
+                "labels": self._generate_labels(),
+            }
+        }
+        if self._flaw.is_embargoed:
+            bts_tracker["fields"]["security"] = {"name": "Embargoed Security Issue"}
+        stream_name = "" if not self._stream else self._stream.name
+        existing_tracker = self._affect.trackers.filter(
+            type=Tracker.TrackerType.JIRA, ps_update_stream=stream_name
```

Comment:
You are right, type would not be necessary here since affect will always have the same component from the same BTS. About the second comment, I believe it is already filtering by the affect itself (`self._affect.trackers`), is that right? I have removed the type, please take a look if now it is correct.

Reply:
You are right that this is already narrowed to a single affect. Therefore the stream is the unique identifier.